### PR TITLE
dnsdist: 1.5.2 -> 1.7.0

### PR DESCRIPTION
**Pull request by jojosch**
###### Motivation for this change
Update to latest release https://dnsdist.org/changelog.html#change-1.7.0. Been using 1.6 and 1.7 it since the `1.6.0-alpha2` on my personal resolvers.

1.7 needs `nghttp2` for outgoing DoH Support (https://dnsdist.org/guides/dns-over-https.html#dns-over-https-doh):

without:
```
$ ./result/bin/dnsdist --version
dnsdist 1.7.0 (Lua 5.2.4)
Enabled features: dns-over-tls(openssl) dns-over-https(DOH) dnscrypt ebpf ipcipher libsodium protobuf re2 recvmmsg/sendmmsg snmp systemd
```

with (note the `outgoing-dns-over-https(nghttp2)`):
```
$ ./result/bin/dnsdist --version
dnsdist 1.7.0 (Lua 5.2.4)
Enabled features: dns-over-tls(openssl) dns-over-https(DOH) dnscrypt ebpf ipcipher libsodium outgoing-dns-over-https(nghttp2) protobuf re2 recvmmsg/sendmmsg snmp systemd
```

----------------------------

Also added a simple NixOS-test and needed to disable 3 tests which require networking:

```
dnsdist> FAIL: testrunner
dnsdist> ================
dnsdist> Running 104 test cases...
dnsdist> Error connecting to new server with address 192.0.2.1:53: connecting socket to 192.0.2.1:53: Network is unreachable
dnsdist> Error connecting to new server with address 192.0.2.2:53: connecting socket to 192.0.2.2:53: Network is unreachable
[...]
dnsdist> Error connecting to new server with address 192.0.2.10:53: connecting socket to 192.0.2.10:53: Network is unreachable
dnsdist> test-dnsdisttcp_cc.cc(213): fatal error: in "test_dnsdisttcp_cc/test_IncomingConnection_BackendNoOOOR": critical check step.request == !d_client ? ExpectedStep::ExpectedRequest::closeClient : ExpectedStep::ExpectedRequest::closeBackend has failed [connect to the backend != close connection to backend]
dnsdist> terminate called after throwing an instance of 'boost::execution_aborted'
dnsdist> unknown location(0): fatal error: in "test_dnsdisttcp_cc/test_IncomingConnection_BackendNoOOOR": signal: SIGABRT (application abort requested)
dnsdist> test-dnsdisttcp_cc.cc(213): last checkpoint
dnsdist> test-dnsdisttcp_cc.cc(213): fatal error: in "test_dnsdisttcp_cc/test_IncomingConnectionOOOR_BackendOOOR": critical check step.request == !d_client ? ExpectedStep::ExpectedRequest::closeClient : ExpectedStep::ExpectedRequest::closeBackend has failed [connect to the backend != close connection to backend]
dnsdist> terminate called recursively
dnsdist> unknown location(0): fatal error: in "test_dnsdisttcp_cc/test_IncomingConnectionOOOR_BackendOOOR": signal: SIGABRT (application abort requested)
dnsdist> test-dnsdisttcp_cc.cc(213): last checkpoint
dnsdist> test-dnsdisttcp_cc.cc(213): fatal error: in "test_dnsdisttcp_cc/test_IncomingConnectionOOOR_BackendNotOOOR": critical check step.request == !d_client ? ExpectedStep::ExpectedRequest::closeClient : ExpectedStep::ExpectedRequest::closeBackend has failed [connect to the backend != close connection to backend]
dnsdist> terminate called recursively
dnsdist> unknown location(0): fatal error: in "test_dnsdisttcp_cc/test_IncomingConnectionOOOR_BackendNotOOOR": signal: SIGABRT (application abort requested)
dnsdist> test-dnsdisttcp_cc.cc(213): last checkpoint
dnsdist> *** 6 failures are detected in the test module "unit"
dnsdist> FAIL testrunner (exit status: 201)
dnsdist> ============================================================================
dnsdist> Testsuite summary for dnsdist 1.6.0
dnsdist> ============================================================================
dnsdist> # TOTAL: 1
dnsdist> # PASS:  0
dnsdist> # SKIP:  0
dnsdist> # XFAIL: 0
dnsdist> # FAIL:  1
dnsdist> # XPASS: 0
dnsdist> # ERROR: 0
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125474` at 4f022de7 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>dnsdist</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/servers/dns/dnsdist/default.nix:49:5:
```
   |
49 |     license = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

**Comment by jojosch** [comment]
Updated to latest release https://github.com/PowerDNS/pdns/releases/tag/dnsdist-1.6.1

**Comment by jojosch** [comment]
Update to latest release https://github.com/PowerDNS/pdns/releases/tag/dnsdist-1.7.0

File: pkgs/servers/dns/dnsdist/default.nix [hunk 1]
```diff
@@ -32,10 +39,14 @@ stdenv.mkDerivation rec {
...
+  passthru.tests = {
+    inherit (nixosTests) dnsdist;
+  };
+
   meta = with lib; {
     description = "DNS Loadbalancer";
     homepage = "https://dnsdist.org";
     license = licenses.gpl2;
-    maintainers = with maintainers; [ SuperSandro2000 ];
+    maintainers = with maintainers; [ SuperSandro2000 jojosch ];
```
Comment (SuperSandro2000) [inline pkgs/servers/dns/dnsdist/default.nix:50]:
```suggestion
    maintainers = with maintainers; [ jojosch ];
```
I can't even remember how I ended up here.

File: nixos/tests/dnsdist.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      };
+      services.dnsdist = {
+        enable = true;
+        listenPort = 5353;
+        extraConfig = ''
+          newServer({address="127.0.0.1:53", name="local-bind"})
+        '';
+      };
+
+      environment.systemPackages = with pkgs; [ dnsutils ];
```
Comment (SuperSandro2000) [inline nixos/tests/dnsdist.nix:35]:
```suggestion
      environment.systemPackages = with pkgs; [ dig ];
```

**Pull request merged**